src/components/app/details/cicdHistory/Sidebar.tsx

```diff
@@ -0,0 +1,273 @@
+import React, { useEffect, useRef } from 'react'
+import { ConditionalWrap, createGitCommitUrl, multiSelectStyles, useIntersection } from '../../../common'
+import { useRouteMatch, useParams, useHistory, generatePath, useLocation } from 'react-router'
+import ReactSelect from 'react-select'
+import { Option, DropdownIndicator } from '../../../v2/common/ReactSelect.utils'
+import moment from 'moment'
+import { Moment12HourFormat, SourceTypeMap } from '../../../../config'
+import { CiPipelineSourceConfig } from '../../../ciPipeline/CiPipelineSourceConfig'
+import {
+  CICDSidebarFilterOptionType,
+    GitTriggers,
+    HistoryComponentType,
+    HistorySummaryCardType,
+    SidebarType,
+    SummaryTooltipCardType,
+} from './types'
+import TippyHeadless from '@tippyjs/react/headless'
+import { NavLink } from 'react-router-dom'
+import { statusColor as colorMap } from '../../config'
+import { ReactComponent as Docker } from '../../../../assets/icons/misc/docker.svg'
+import ReactGA from 'react-ga4'
+import DetectBottom from '../../../common/DetectBottom'
+const Sidebar = React.memo(({ type, filterOptions, triggerHistory, hasMore, setPagination }: SidebarType) => {
+    const { pipelineId, appId, envId } = useParams<{ appId: string; envId: string; pipelineId: string }>()
+    const { push } = useHistory()
+    const { path } = useRouteMatch()
+    const handleFilterChange = (selectedFilter: CICDSidebarFilterOptionType): void => {
+        if (type === HistoryComponentType.CI) {
+          setPagination({ offset: 0, size: 20 })
+            push(generatePath(path, { appId, pipelineId: selectedFilter.value }))
+        } else {
+          setPagination({ offset: 0, size: 20 })
+            push(generatePath(path, { appId, envId: selectedFilter.value,  pipelineId: selectedFilter.pipelineId }))
+        }
+    }
+    function reloadNextAfterBottom() {
+        ReactGA.event({
+            category: 'pagination',
+            action: 'scroll',
+            label: `${type.toLowerCase()}-history`,
+            value: triggerHistory.size,
+        })
+        setPagination({ offset: triggerHistory.size, size: 20 })
+    }
+    const selectedFilter = filterOptions?.find(
+        (filterOption) => filterOption.value === (type === HistoryComponentType.CI ? pipelineId : envId),
+    )
+    return (
+        <>
+            <div className="select-pipeline-wrapper w-100 pl-16 pr-16" style={{ overflow: 'hidden' }}>
```

Comment:
use class `dc__overflow-hidden` instead of inline style for overflow

Reply:
Done